x/dsmr/node_test.go

```diff
@@ -435,33 +435,110 @@ func TestNode_AcceptedChunksAvailableOverGetChunk(t *testing.T) {
 
 // Node should be willing to sign valid chunks
 func TestNode_GetChunkSignature_SignValidChunk(t *testing.T) {
+	nodeID := ids.GenerateTestNodeID()
+	sk, err := bls.NewSecretKey()
+	require.NoError(t, err)
+	pk := bls.PublicFromSecretKey(sk)
 	tests := []struct {
-		name     string
-		verifier Verifier[dsmrtest.Tx]
-		wantErr  error
+		name                      string
+		verifier                  Verifier[dsmrtest.Tx]
+		producerNode              ids.NodeID
+		wantErr                   error
+		nodeLastAcceptedTimestamp int64
+		chunkExpiry               int64
 	}{
 		{
-			name:     "invalid chunk",
-			verifier: failVerifier{},
-			wantErr:  ErrInvalidChunk,
+			name:                      "invalid chunk",
+			verifier:                  failVerifier{},
+			wantErr:                   ErrInvalidChunk,
+			producerNode:              ids.GenerateTestNodeID(),
+			nodeLastAcceptedTimestamp: 1,
+			chunkExpiry:               123,
 		},
 		{
-			name: "valid chunk",
-			verifier: ChunkVerifier[dsmrtest.Tx]{
-				networkID: networkID,
-				chainID:   chainID,
-			},
+			name: "invalid chunk ( bad producer id )",
+			verifier: NewChunkVerifier[dsmrtest.Tx](
+				networkID,
+				chainID,
+				pChain{
+					validators: []Validator{{
+						NodeID:    nodeID,
+						Weight:    1,
+						PublicKey: pk,
+					}},
+				},
+				1,
+				1,
+			),
+			wantErr:                   ErrInvalidChunk,
```

Comment:
Can we use more specific errors in the code and assertions in the test so that we have higher confidence it's failing for the expected reason?

Reply:
yes